modules/sort/test/spec.coffee

```diff
@@ -108,3 +141,52 @@ describe 'hx-sort tests', ->
         if supportsOptions()
           array.sort(hx.sort.localeCompare('sv')).should.eql(['a', 'ä', 'e', 'é', 'è', 'z'])
           array.sort(hx.sort.localeCompare('de')).should.eql(['a', 'e', 'é', 'è', 'z', 'ä'])
+
+  describe 'using localeCompare potentially with Intl.Collator. ', localeCompareTests
+
+  describe 'using localeCompare without Intl.Collator. ', runWithoutCollator(localeCompareTests)
+
+  compareNullsLastTests = ->
+    it 'Strings should return the right value', ->
+      hx.sort.compareNullsLast('a', 'b').should.equal(-1)
+      hx.sort.compareNullsLast('b', 'a').should.equal(1)
+      hx.sort.compareNullsLast('a', 'a').should.equal(0)
+
+    it 'undefined should always be sorted to the end', ->
+      hx.sort.compareNullsLast(undefined, 'a').should.equal(1)
+      hx.sort.compareNullsLast('a', undefined).should.equal(-1)
+      hx.sort.compareNullsLast(undefined, 'z').should.equal(1)
+      hx.sort.compareNullsLast('z', undefined).should.equal(-1)
+      hx.sort.compareNullsLast(undefined, null).should.equal(1)
+      hx.sort.compareNullsLast(null, undefined).should.equal(-1)
+      hx.sort.compareNullsLast(undefined, undefined).should.equal(0)
+
+    it 'null should be sorted after any defined value', ->
+      hx.sort.compareNullsLast(null, 'a').should.equal(1)
+      hx.sort.compareNullsLast('a', null).should.equal(-1)
+      hx.sort.compareNullsLast(null, 'z').should.equal(1)
+      hx.sort.compareNullsLast('z', null).should.equal(-1)
+      hx.sort.compareNullsLast(null, null).should.equal(0)
+
+    it 'Arrays should return the right value', ->
+      array = ['c', 'a', 'b', 'aa', 'ab', 'ac', '1', '2']
+      array.sort(hx.sort.compareNullsLast).should.eql(['1', '2', 'a', 'aa', 'ab', 'ac', 'b', 'c'])
+      array = ['25', '10', '20', '42', '100']
+      array.sort(hx.sort.compareNullsLast).should.eql(['10', '20', '25', '42', '100'])
+
+    it 'Arrays of strings with undefined values should sort null and undefined to the end', ->
+      array = ['b', undefined, null, 'z', 'a', 'p']
+      array.sort(hx.sort.compareNullsLast).should.eql(['a', 'b', 'p', 'z', null, undefined])
```

Comment:
could you add multiple `undefined` and `null` to the array? Just for sanity check on `if a is b`